Entities hibernate

MalishkaMyu opened the pull request.

MalishkaMyu committed:
Entities creation and rename tables

1) Creation 3 entities: Role, Department, Employee
2) Creation Enum Gender and GenderConverter for the filed sex
3) rename tables (necessary for ddl-auto=validate)
4) modify id-columns type int->bigint (necessary for ddl-auto=validate)

AlecKotovichSAM commented on src/main/java/com/samsolutions/employeesdep/model/entities/Department.java:
```diff
@@ -0,0 +1,38 @@
+    @Column(name = "depart_id", unique = true, nullable = false)
+    @SequenceGenerator(name = "pk_seq_depart", sequenceName = "seq_depart_id", allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "pk_seq_depart")
+    private long id;
```
Better use Long instead. Otherwise there will be a problem with nulls

AlecKotovichSAM commented on src/main/java/com/samsolutions/employeesdep/model/entities/Employee.java:
```diff
@@ -0,0 +1,122 @@
+    @Column(name = "emp_id", unique = true, nullable = false)
+    @SequenceGenerator(name = "pk_seq_emp", sequenceName = "seq_emp_id", allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "pk_seq_emp")
+    private long id;
```
long -> Long

AlecKotovichSAM commented on src/main/resources/dbscripts/V1.3__change_id_employee_roles.sql:
```diff
@@ -0,0 +1,7 @@
+BEGIN;
+
+ALTER TABLE IF EXISTS public."employee_roles"
```
Rename employee_roles -> employees_roles

AlecKotovichSAM reviewed (changes requested):
Conversations

MalishkaMyu committed:
Entities creation and rename tables

1) Creation 3 entities: Role, Department, Employee
2) Creation Enum Gender and GenderConverter for the filed sex
3) rename tables (necessary for ddl-auto=validate)
4) modify id-columns type int->bigint (necessary for ddl-auto=validate)

MalishkaMyu commented:
I've commited changes, not pushed. Is it enough?
